src/kzg10/mod.rs

```diff
@@ -191,6 +218,30 @@ where
         Ok((Commitment(commitment.into()), randomness))
     }
 
+    /// Outputs a commitment to `polynomial` in G2
+    pub fn G2_commit(
```

Comment:
We should also rename `commit` to `commit_g1`.

```suggestion
    pub fn commit_g2(
```

Reply:
This part is also a little awkward, because to `check` a KZG proof you have to commit the witness/evaluation polynomial in G1 and the subproduct domain polynomial in G2, both of which in the case of a single reveal can happen implicitly, but with a multi reveal should probably call `commit_g1` and `commit_g2` which requires the `UniversalParams`. 

Or, the `VerifierKey` needs to include a lot more, and actually might need to include more powers than the committer key (in the case where the multi reveal domain is larger than the degree of the committed polynomial)

Reply:
> Or, the VerifierKey needs to include a lot more, and actually might need to include more powers than the committer key (in the case where the multi reveal domain is larger than the degree of the committed polynomial)

That's fine, as long as we make it parameterized.